### PR TITLE
[sdks] Add mac-{mac32,mac64} targets

### PR DESCRIPTION
**Pull request by luhenry**

### CODE REVIEW

File: autogen.sh [hunk 1]
```diff
@@ -120,7 +120,7 @@ for PARAM; do
 done
 
 if test x$has_ext_mod = xtrue; then
-	pushd ../mono-extensions/scripts
+	pushd $top_srcdir../mono-extensions/scripts
```
Comment (lewurm) [inline autogen.sh]: shouldn't that be `$top_srcdir/../mono-extensions/scripts`?

Comment (luhenry) [reply autogen.sh]: No, `$top_srcdir` is always `/`-terminated.

File: scripts/ci/run-jenkins.sh [hunk 2]
```diff
@@ -145,9 +145,14 @@ if [[ ${CI_TAGS} == *'sdks-ios'* ]];
 	   if [[ ${CI_TAGS} == *'cxx'* ]]; then
 		   echo "ENABLE_CXX=1" >> sdks/Make.config
 	   fi
+        if [[ ${CI_TAGS} != *'debug'* ]]; then
+            echo "CONFIGURATION=debug"
+        fi
```
Comment (lewurm) [inline scripts/ci/run-jenkins.sh]: indent mismatch

File: scripts/ci/run-jenkins.sh [hunk 3]
```diff
@@ -145,9 +145,14 @@ if [[ ${CI_TAGS} == *'sdks-ios'* ]];
 	   if [[ ${CI_TAGS} == *'cxx'* ]]; then
 		   echo "ENABLE_CXX=1" >> sdks/Make.config
 	   fi
+        if [[ ${CI_TAGS} != *'debug'* ]]; then
```
Comment (akoeplinger) [inline scripts/ci/run-jenkins.sh]: I assume this should be `==` instead of `!=` right?

File: scripts/ci/run-jenkins.sh [hunk 4]
```diff
@@ -219,16 +227,23 @@ if [[ ${CI_TAGS} == *'webassembly'* ]] || [[ ${CI_TAGS} == *'wasm'* ]];
    then
 	   echo "DISABLE_ANDROID=1" > sdks/Make.config
 	   echo "DISABLE_IOS=1" >> sdks/Make.config
+	   echo "DISABLE_MAC=1" >> sdks/Make.config
 	   echo "DISABLE_DESKTOP=1" >> sdks/Make.config
 	   if [[ ${CI_TAGS} == *'cxx'* ]]; then
 	       echo "ENABLE_CXX=1" >> sdks/Make.config
 	   fi
+        if [[ ${CI_TAGS} != *'debug'* ]]; then
+            echo "CONFIGURATION=debug"
```
Comment (lewurm) [inline scripts/ci/run-jenkins.sh]: either that should be
```sh
export CONFIGURATION=debug
```
or
```sh
echo "CONFIGURATION=debug" >> sdks/Make.config
```

File: sdks/builds/Makefile [hunk 5]
```diff
@@ -44,7 +45,7 @@ all: package
 configure-mono: $(TOP)/configure
 
 $(TOP)/configure: $(TOP)/configure.ac $(TOP)/autogen.sh
-	cd $(TOP) && PATH=$(EXTRA_PATH):$$PATH NOCONFIGURE=1 ./autogen.sh
+	cd $(TOP) && PATH=$(EXTRA_PATH):$$PATH NOCONFIGURE=1 ./autogen.sh $(if $(wildcard $(TOP)/../mono-extensions),--enable-extension-module=xamarin --enable-extension-module)
```
Comment (akoeplinger) [inline sdks/builds/Makefile]: afaik we don't need `--enable-extension-module=xamarin`, just `--enable-extension-module`.

Comment (luhenry) [reply sdks/builds/Makefile]: We need both, see `mono-extension` repo for the rational.

File: scripts/ci/run-jenkins.sh (hunk 4 above)
Comment (luhenry) [reply scripts/ci/run-jenkins.sh]: The second one.

File: sdks/builds/mac.mk [hunk 6]
```diff
@@ -0,0 +1,51 @@
+
+mac_PLATFORM_BIN=$(XCODE_DIR)/Toolchains/XcodeDefault.xctoolchain/usr/bin
+
+##
+# Parameters
+#  $(1): target
```
Comment (lewurm) [inline sdks/builds/mac.mk]: `$(2)` isn't documented

**Comment by luhenry** [comment]
@monojenkins backport to 2019-02
@monojenkins backport to 2018-12
@monojenkins backport to 2018-10

**Pull request merged**